app/api/threads/list_threads.go

```diff
@@ -70,7 +84,41 @@ type participant struct {
 //		- name: is_mine
 //			in: query
 //			type: boolean
-//
+//		- name: status
+//			description: Filter by status
+//			in: query
+//			type: string
+//			enum: [waiting_for_participant,waiting_for_trainer,closed]
+//		- name: latest_update_gt
+//			description: Only threads where `latest_update_at`>`latest_update_gt`.
+//			in: query
+//			type: string
+//			format: date-time
+//		- name: sort
+//			in: query
+//			default: [-latest_update_at]
+//			type: array
+//			items:
+//				type: string
+//				enum: [latest_update_at,-latest_update_at]
```

Comment:
Sort must always ends with tie-breaker (basically an unique key for the listed entity), which should be part of the "default" sort, be part of the enum, and automatically added to the sort if not present.
Otherwise, the output is deterministic.

Reply:
After checking, the pagination system it is already appending the "TieBreakers" parameters to the default sort:
`ORDER BY threads.latest_update_at DESC, items.id ASC, threads.participant_id ASC LIMIT 500`

It is added if no sort parameter is used (default), or when `&sort=latest_update_at` is used.

The only thing done was adding the TieBreakers parameter here:
```
	queryDB, apiError = service.ApplySortingAndPaging(r, queryDB, &service.SortingAndPagingParameters{
		Fields: service.SortingAndPagingFields{
			"latest_update_at": {ColumnName: "threads.latest_update_at"},
			"item_id":          {ColumnName: "items.id"},
			"participant_id":   {ColumnName: "threads.participant_id"},
		},
		DefaultRules: "-latest_update_at",
		TieBreakers: service.SortingAndPagingTieBreakers{
			"item_id":        service.FieldTypeInt64,
			"participant_id": service.FieldTypeInt64,
		},
	})
```

Is it still necessary to add item_id and participant_id in the sort paramter?

Reply:
fixed the doc